packages/paste-core/components/badge/src/index.tsx

```diff
@@ -1,49 +1,70 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import {Box} from '@twilio-paste/box';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {secureExternalLink} from '@twilio-paste/anchor';
+import {ErrorIcon} from '@twilio-paste/icons/esm/ErrorIcon';
 
 import type {BadgeProps, BadgeVariants} from './types';
 import {useResizeChildIcons} from './hooks';
-import {badgeFocusableStyles, badgeVariantStyles} from './styles';
-import {getBadgeSpanProps, isFocusableElement} from './utils';
-import {BadgeWrapper} from './BadgeWrapper';
+import {badgeVariantStyles, getBadgeAnchorStyles, getBadgeButtonStyles} from './styles';
 
 export type {BadgeProps};
 
 export const Badge = React.forwardRef<HTMLElement, BadgeProps>(
-  ({variant, children, element = 'BADGE', ...props}, ref) => {
+  ({as, href, variant, children, element = 'BADGE', ...props}, ref) => {
     const resizedChildren = useResizeChildIcons(children);
 
-    const variantStyles = badgeVariantStyles[variant];
-    const spanProps = getBadgeSpanProps(props);
-    const isFocusable = isFocusableElement(props);
+    let badgeStyles = badgeVariantStyles[variant];
+
+    if (as != null) {
```

Comment:
Question: because as is required, it will never be null or undefined.  Did you add this because foa typescript error?

Reply:
oh great question! I added this because of a ts error before we changed the types overall. Since we've changed them, this does work when I take it away. Thanks Lee!